client/components/advanced-filters/index.js

```diff
@@ -0,0 +1,215 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { Component, Fragment } from '@wordpress/element';
+import { SelectControl, Button, FormTokenField, Dropdown } from '@wordpress/components';
+import { partial, findIndex, find, difference } from 'lodash';
+import PropTypes from 'prop-types';
+import Gridicon from 'gridicons';
+
+/**
+ * Internal dependencies
+ */
+import Card from 'components/card';
+import './style.scss';
+
+class AdvancedFilters extends Component {
+	constructor( props ) {
+		super( props );
+		this.state = {
+			match: props.config.matches[ 0 ],
+			activeFilters: [
+				/**
+				 * Example activeFilter
+				 * { key: ‘product’, rule: ‘includes’, value: [ ‘one’, ‘two’ ] }
+				 */
+			],
+		};
+
+		this.onMatchChange = this.onMatchChange.bind( this );
+		this.onFilterChange = this.onFilterChange.bind( this );
+		this.getSelector = this.getSelector.bind( this );
+		this.getAvailableFilterKeys = this.getAvailableFilterKeys.bind( this );
+		this.addFilter = this.addFilter.bind( this );
+		this.removeFilter = this.removeFilter.bind( this );
+		this.clearAll = this.clearAll.bind( this );
+	}
+
+	onMatchChange( value ) {
+		this.setState( {
+			match: find( this.props.config.matches, match => value === match.value ),
+		} );
+	}
+
+	onFilterChange( key, property, value ) {
+		const activeFilters = this.state.activeFilters.map( activeFilter => {
+			if ( key === activeFilter.key ) {
+				return Object.assign( {}, activeFilter, { [ property ]: value } );
+			}
+			return activeFilter;
+		} );
+
+		this.setState( { activeFilters } );
+	}
+
+	removeFilter( key ) {
+		const activeFilters = [ ...this.state.activeFilters ];
+		const index = findIndex( activeFilters, f => f.key === key );
+		activeFilters.splice( index, 1 );
+		this.setState( { activeFilters } );
+	}
+
+	getTitle() {
+		const { config } = this.props;
+		const { match } = this.state;
+		const label = sprintf( __( 'Orders Match %s Filters', 'wc-admin' ), match.label );
```

Comment:
Will this label be configurable via a prop? Guessing we will use this on other reports on than those that list orders? Or just make it more generic like "Items Match %s Filters"